File: checks/src/main/java/com/zhenai/lib/checks/CheckList.java
/*
 * SonarSource SLang
 * Copyright (C) 2018-2019 SonarSource SA
 * mailto:info AT sonarsource DOT com
 *
 * This program is free software; you can redistribute it and/or
 * modify it under the terms of the GNU Lesser General Public
 * License as published by the Free Software Foundation; either
 * version 3 of the License, or (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
 * Lesser General Public License for more details.
 *
 * You should have received a copy of the GNU Lesser General Public License
 * along with this program; if not, write to the Free Software Foundation,
 * Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
 */
package com.zhenai.lib.checks;

import com.zhenai.lib.checks.api.ICheck;

import java.util.Arrays;
import java.util.List;

public class CheckList {
    private CheckList() {
    }

    public static List<ICheck> allChecks() {
        return Arrays.asList(
                new TodoCommentCheck(),
                new TooManyParametersCheck(),
                new UnusedFunctionParameterCheck(),
                new HardCodeStringCheck(),
                new UseLogCheck(),
                new ToastCheck(),
                new EmptyCommentCheck(),
                new InterfaceMethodCommentCheck(),
                new UnusedPrivateMethodCheck(),
                new UnusedLocalVariableCheck(),
                new ObjectDeclarationCheck(),
                new TooComplexExpressionCheck(),
                new TooLongFunctionCheck(),
                new TooManyLinesOfCodeFileCheck(),
                new CommentedCodeCheck(),
                new EmptyBlockCheck(),
                new DuplicateBranchCheck(),
                new EmptyFunctionCheck());
    }

    public static ICheck getSlangCheck(String name) {
        for (ICheck check : allChecks()) {
            if (check.getSIssue().getName().equals(name)) {
                return check;
            }
        }
        return null;
    }
}

File: checks/src/main/kotlin/com/zhenai/lib/checks/CheckList.kt
package com.zhenai.lib.checks

import com.zhenai.lib.checks.api.ICheck
import com.zhenai.lib.checks.comments.CommentedCodeCheck
import com.zhenai.lib.checks.comments.EmptyCommentCheck
import com.zhenai.lib.checks.comments.InterfaceMethodCommentCheck
import com.zhenai.lib.checks.complexity.*
import com.zhenai.lib.checks.empty.EmptyBlockCheck
import com.zhenai.lib.checks.empty.EmptyFunctionCheck
import com.zhenai.lib.checks.other.*
import com.zhenai.lib.checks.performance.ArrayPrimitiveCheck
import com.zhenai.lib.checks.performance.LazyCheck
import com.zhenai.lib.checks.performance.ObjectDeclarationCheck
import com.zhenai.lib.checks.unuse.UnusedFunctionParameterCheck
import com.zhenai.lib.checks.unuse.UnusedLocalVariableCheck
import com.zhenai.lib.checks.unuse.UnusedPrivateMethodCheck
import java.util.*

class CheckList {

    private constructor()

    companion object {

        fun allChecks(): List<ICheck> {
            return Arrays.asList(
                TodoCommentCheck(),
                TooManyParametersCheck(),
                UnusedFunctionParameterCheck(),
                HardCodeStringCheck(),
                UseLogCheck(),
                ToastCheck(),
                EmptyCommentCheck(),
                InterfaceMethodCommentCheck(),
                UnusedPrivateMethodCheck(),
                UnusedLocalVariableCheck(),
                ObjectDeclarationCheck(),
                TooComplexExpressionCheck(),
                TooLongFunctionCheck(),
                TooManyLinesOfCodeFileCheck(),
                CommentedCodeCheck(),
                EmptyBlockCheck(),
                DuplicateBranchCheck(),
                LazyCheck(),
                VariableAndParameterNameCheck(),
                ArrayPrimitiveCheck(),
                EmptyFunctionCheck()
            )
        }

        fun getSlangCheck(name: String): ICheck? {
            for (check in allChecks()) {
                if (check.getSIssue().name == name) {
                    return check
                }
            }
            return null
        }
    }

}